pkg/addonmanager/controllers/agentdeploy/utils.go

```diff
@@ -492,3 +499,32 @@ func getDeletionOrphaningRule(obj runtime.Object) (*workapiv1.OrphaningRule, err
 	}
 	return rule, nil
 }
+
+func configsToAnnotations(configReference []addonapiv1alpha1.ConfigReference) (map[string]string, error) {
+	if len(configReference) == 0 {
+		return nil, nil
+	}
+
+	annotations := make(map[string]string, len(configReference))
+	for _, v := range configReference {
+		if v.DesiredConfig == nil {
+			continue
+		}
+		resourceStr := v.Resource
+		if len(v.Group) > 0 {
+			resourceStr += fmt.Sprintf(".%s", v.Group)
```

Comment:
we need a prefix which is common. the function also need a doc on what annotation will be generated.